### ISSUE TITLE
[Portal, Authentication] The search behaviour in Phone Number section is not accurate

### ISSUE THREAD

**Issue by joyzoursky** [authgear/authgear-server#639]
## Environment:
- Platform: Web Portal
- Browser: Chrome on Mac

## Description:
Currently the search results returned by the phone no. searching is strange, i.e. searching 'ho' will return 3 results, while searching 'hon' will return ~20 results.

## Steps to reproduce:
1. Log in to the Authgear portal
2. Click into any app entity
3. Go to the 'Authentication' and enable the 'Phone Number' field
4. Search 'ho' and 'hon' and see the results returned does not contain the exact keyword

## Expected result:
Should search by exact keyword

## Screenshot:
![image](https://gitlab.com/skytester/authgear-server/uploads/8655cd55e5169e8b40d65561caf825dc/image.png)
![image](https://gitlab.com/skytester/authgear-server/uploads/7d8ab2678ec18c3062d70acf7f8ada4d/image.png)

**Comment by axolotl0212** [comment]
@louischan-oursky @kiootic I think I have figured out why this happens.

As the score uses normalized `Levenshtein distance`, more letter means the denominator is larger. We have a threshold to determine which item is a match, say we use 0.4 as a threshold (lower is better):

- If there is 2 letters, 1 letter mismatch, the normalized distance is 0.5, so only item with both letter no mismatch letter is a match.
- If there is 3 letters, 1 letter mismatch, the normalized distance is 0.333, so 1 letter mismatch is allowed

Back to this example, when search string is "ho", we require the item to contain "ho". When search string is "hon", we allow 1 mismatch, so there is 4 possible combination ("ho?", "h?n", "?on", "hon") where "?" can be empty or any letter. Hence there is more matching items.

So in conclusion, this is not related to the specific library, it is caused by the threshold and normalization which is expected behaviour

**Comment by kiootic** [comment]
Maybe use n-gram search instead?

**Comment by axolotl0212** [comment]
> Maybe use n-gram search instead?

N-gram search does not allow mismatch character? I tried to search for some library but they are pretty old (>1 year). I think using fuzzy search with threshold ~ 0 distance or just `"string".includes("str")` has similar effect?

**Comment by kiootic** [comment]
I don't mind changing it to use `String.includes` instead.

**Comment by louischan-oursky** [comment]
Let's use switch to use `toLowerCase()` and `includes()`
